extensions/positron-r/src/languageFeatures/rFilePasteProvider.ts

```diff
@@ -0,0 +1,71 @@
+/*---------------------------------------------------------------------------------------------
+ *  Copyright (C) 2025 Posit Software, PBC. All rights reserved.
+ *  Licensed under the Elastic License 2.0. See LICENSE.txt for license information.
+ *--------------------------------------------------------------------------------------------*/
+
+import * as vscode from 'vscode';
+import * as positron from 'positron';
+
+/**
+ * Document paste edit provider for R files that converts files on the clipboard
+ * into file paths that are usable in R code.
+ */
+export class RFilePasteProvider implements vscode.DocumentPasteEditProvider {
+
+	/**
+	 * Provide paste edits for R filepaths when files are detected on clipboard.
+	 */
+	async provideDocumentPasteEdits(
+		document: vscode.TextDocument,
+		ranges: readonly vscode.Range[],
+		dataTransfer: vscode.DataTransfer,
+		context: vscode.DocumentPasteEditContext,
+		token: vscode.CancellationToken
+	): Promise<vscode.DocumentPasteEdit[] | undefined> {
+
+		const setting = vscode.workspace.getConfiguration('positron.r').get<boolean>('autoConvertFilePaths');
+		if (!setting) {
+			return undefined;
+		}
+
+		const filePaths = await positron.paths.extractClipboardFilePaths(dataTransfer, {
+			preferRelative: true
+		});
+
+		if (!filePaths) {
+			return undefined;
+		}
+
+		// Format for R: single path or R vector syntax
+		const insertText = filePaths.length === 1
+			? filePaths[0]
+			: `c(${filePaths.join(', ')})`;
+
+		// TODO @jennybc: at some point, this sort of special paste WAS showing
+		// up when using the "Paste As..." command. Now it doesn't. Presumably
+		// had something to with the code I've since ripped out. Would be nice
+		// to get that back.
+		return [{
+			insertText,
+			title: 'Insert file path(s)',
```

Comment:
Should this text be localized?

Reply:
At the moment, I don't think this text is ever surfaced 🫠 At some point, when this PR "did more" but also had a weaker design, this text was surfaced (in a "Paste As" sort of experience). I will rationalize this bit one way or another before I merge.